lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1428,6 +1498,35 @@ ParseResult FIRStmtParser::parseOptionalExpPostscript(Value &result) {
   }
 }
 
+template <typename subop>
+FailureOr<Value>
+FIRStmtParser::emitCachedSubAccess(Value base, ArrayRef<NamedAttribute> attrs,
```

Comment:
should we be caching and CSEing things in the parser?

Reply:
We've gone both ways on this.  Connect expands eagerly, other things do a straight parse.  The argument for the latter often came down to whether an annotation might target it.

Reply:
Interesting.  This does add new caching for the ref.sub's but otherwise is a refactoring of the caching pattern duplicated around, for re-use when indexing through references.